### PR TITLE
[WIP] Initial sketch of a CLI for ytree

### PR DESCRIPTION
**Pull request by matthewturk**
This is an initial sketch of a command line interface for ytree.  At present, note that because it pretty-prints, not everything will be displayed when you ask for an array.  Ultimately the format returned will be made flexible.

As of opening the PR, these commands work:

```#bash

$ ytree info merger_trees/normal_bg/normal_bg.h5
{'box_size': 28.4 Mpc/h,
 'derived_field_list': ['redshift'],
 'field_list': ['scale_factor',
                'uid',
                'desc_scale',
                'desc_uid',
[snip]
                'BH_Mass'],
 'omega_lambda': 0.73399999999999999,
 'omega_matter': 0.26600000000000001,
 'tree_count': 37733}

$ ytree tree_info merger_trees/normal_bg/normal_bg.h5 0 mass 
2530985915.492958 Msun
```

I'll be working on this a bit, but any feedback at this stage would be appreciated!  I'm aiming to make the return method more flexible as well as add more commands.

### CODE REVIEW

**Comment by manodeep** [comment]
Ooo this looks cool :) 

Frequently, we need to retrieve some quantity, say `star formation rate`, along the entire progenitor history for one galaxy (i.e., retrieve `sfr` values for all other galaxies that eventually merge with this particular galaxy). Would the `cli` be able to handle something like that?

**Comment by MatthewTurk** [comment]
I hope so! Querying on progenitors is next up.

File: ytree/utilities/command_line.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import argparse
+import ytree
+import pprint
+
+def statistics(args):
+    arbor = ytree.load(args.arbor_file)
+    return { 'box_size': arbor.box_size,
+             'omega_lambda': arbor.omega_lambda,
+             'omega_matter': arbor.omega_matter,
+             'tree_count': arbor.trees.size,
```
Comment (brittonsmith) [inline ytree/utilities/command_line.py]: Maybe `hubble_constant` too?

**Comment by brittonsmith** [comment]
@MatthewTurk, I think it would be very useful to query based on redshift, time, or really any other field since they're all treated the same way. Currently, there is the following functionality, which I need to document:
```
my_halos = a.select_halos('tree["tree", "redshift"] < 20',
                          fields=["redshift"])
```
You might be able to utilize that for these purposes, or even improve on that. Currently, it works on `eval` syntax, which is a bit clunky but does the trick. I was hoping to switch this out for a ytree frontend in `yt` that makes use of the demeshening when that's ready.

Either way, this ability to query for halos either in a single tree or the whole arbor would be super useful.

**Comment by MatthewTurk** [comment]
Would you be open to a separate pull request that implements the redshift query without `eval`?  I'm somewhat reluctant to expose anything that gets `eval`'d through the CLI, because I want this to become consumed through a REST interface eventually.

**Comment by MatthewTurk** [comment]
Sorry, let me rephrase -- would you be open to a reasonably simple change that avoided `eval` in favor of a simple "less than" "greater than" "equal to" syntax for selecting on one or more fields?  That way it wouldn't be specific to the redshift, but would allow selection without `eval`.

**Comment by brittonsmith** [comment]
That makes total sense to me. I would be fine with a bespoke function that avoids`eval` statements.

**Comment by manodeep** [comment]
@MatthewTurk Would it be possible to have an option where selection is made on multiple fields? The immediate use-case that comes to mind is finding massive, quiscent galaxies at high-z. The selections would be 'M_star > some number, z > another number'. 

For the tree selection, we also need to compute SEDs and for that we need to retrieve the star formation history of all progenitor galaxies. Then, if possible, an option to get **all** progenitors of a target object would be fantastic.

**Comment by brittonsmith** [comment]
@matthewturk, I'm just going through long outstanding PRs. Would you like me to keep this open?

**Comment by brittonsmith** [comment]
@matthewturk, I'm going to close this for now, but I would like to revisit it after search and selection have been refactored. To be continued then!
